pkgs/development/interpreters/clojurescript/lumo/default.nix

```diff
@@ -0,0 +1,257 @@
+      lumo.repl \
+      lumo.repl-resources \
+      lumo.js-deps \
+      lumo.common '';
+
+    compileClojurescript = (simple: ''
+      (require '[cljs.build.api :as cljs])
+      (cljs/build \"src/cljs/snapshot\"
+        {:optimizations      ${if simple then ":simple" else ":none"}
+         :main               'lumo.core
+         :cache-analysis     true
+         :source-map         false
+         :dump-core          false
+         :static-fns         true
+         :optimize-constants false
+         :npm-deps           false
+         :verbose            true
+         :closure-defines    {'cljs.core/*target*       \"nodejs\"
+                              'lumo.core/*lumo-version* \"${version}\"}
+         :compiler-stats     true
+         :process-shim       false
+         :fn-invoke-direct   true
+         :parallel-build     false
+         :browser-repl       false
+         :target             :nodejs
+         :hashbang           false
+         ;; :libs               [ \"src/cljs/bundled\" \"src/js\" ]
+         :output-dir         ${if simple
+                                    then ''\"cljstmp\"''
+                                  else ''\"target\"''}
+         :output-to          ${if simple
+                                    then ''\"cljstmp/main.js\"''
+                                  else ''\"target/deleteme.js\"'' }})
+    '');
+
+
+    cacheToJsons = ''
+      (import [java.io ByteArrayOutputStream FileInputStream])
+      (require '[cognitect.transit :as transit]
+               '[clojure.edn :as edn]
+               '[clojure.string :as str])
+
+      (defn write-transit-json [cache]
+        (let [out (ByteArrayOutputStream. 1000000)
+              writer (transit/writer out :json)]
+          (transit/write writer cache)
+          (.toString out)))
+
+      (defn process-caches []
+        (let [cache-aot-path      \"target/cljs/core.cljs.cache.aot.edn\"
+              cache-aot-edn       (edn/read-string (slurp cache-aot-path))
+              cache-macros-path   \"target/cljs/core\$macros.cljc.cache.json\"
+              cache-macros-stream (FileInputStream. cache-macros-path)
+              cache-macros-edn    (transit/read (transit/reader cache-macros-stream :json))
+              caches              [[cache-aot-path cache-aot-edn]
+                                   [cache-macros-path cache-macros-edn]]]
+          (doseq [[path cache-edn] caches]
+            (doseq [key (keys cache-edn)]
+              (let [out-path (str/replace path #\"(\.json|\.edn)\$\"
+                               (str \".\" (munge key) \".json\"))
+                    tr-json  (write-transit-json (key cache-edn))]
+                (spit out-path tr-json))))))
+
+      (process-caches)
+    '';
+
+    trimMainJsEnd = ''
+      (let [string (slurp \"target/main.js\")]
+        (spit \"target/main.js\"
+          (subs string  0 (.indexOf string \"cljs.nodejs={};\"))))
+    '';
+
+
+    cljdeps   = import ./deps.nix;
```

Comment:
Then change this to `cljdeps = import ./deps.nix pkgs;` while getting `pkgs` by adding it to the argument list at the top.

Reply:
good idea, I do that!